### PR TITLE
NOX-29 entitiy translations Create/Update/Read

### PR DESCRIPTION
**Pull request by msadikkoseaupa**

### CODE REVIEW

File: src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Dto/WorkplaceLocalizedCreateDto.g.cs
Comment (rochar) [inline src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Dto/WorkplaceLocalizedCreateDto.g.cs:1]: shoud we not have a single `<EntityName>LocalizedUpsertDto`?

File: src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Queries/GetWorkplaceTranslationsByIdQuery.g.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+using Microsoft.EntityFrameworkCore;
+
+using Nox.Application.Commands;
+
+using ClientApi.Application.Dto;
+using ClientApi.Infrastructure.Persistence;
+
+namespace ClientApi.Application.Queries;
+
+public record GetWorkplaceTranslationsByIdQuery(System.UInt32 keyId, System.String CultureCode) : IRequest <IQueryable<WorkplaceLocalizedDto>>;
```
Comment (rochar) [inline src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Queries/GetWorkplaceTranslationsByIdQuery.g.cs:15]: CultureCode must be a Nox.Type.CultureCode, so we benefit from intention and have validations OOB

File: src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Domain/WorkplaceLocalized.g.cs [hunk 2]
```diff
@@ -13,11 +13,11 @@
 using Nox.Types;
 
 namespace ClientApi.Domain;
-
 /// <summary>
 /// Workplace.
 /// </summary>
-internal partial class WorkplaceLocalized : IEntityConcurrent
+[PrimaryKey(nameof(Id),nameof(CultureCode))]
```
Comment (rochar) [inline src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Domain/WorkplaceLocalized.g.cs:19]: we are using the modelcreating in dbcontext to configure dto, we are not using attributes for this, I would prefer to keep consistency on this

File: src/Nox.ClientApi.Tests/Tests/Controllers/WorkplacesControllerTests.cs [hunk 3]
```diff
@@ -350,6 +353,45 @@ public async Task Post_WhenInvokedMultipleTimes_CreatesCorrectLocalizations()
...
+            // Act
+            var result = await PostAsync<WorkplaceCreateDto, WorkplaceDto>(Endpoints.WorkplacesUrl, createDto, headers);
+
+            var localizationCultureCode = "tr-TR";
+            var createLocalizedDto = new WorkplaceLocalizedCreateDto
+            {
+               Description = "Gare de Lyon ve Gare d'Austerlitz'e sadece birkaç dakika uzaklıkta, park yeri olan modern, mütevazı büyüklükte bir bina.",
+            };
+
+            var localizedNew = GetEntitiesByFilter<WorkplaceLocalized>(dto => dto.Id == Nuid.From(result!.Id));
```
Comment (rochar) [inline src/Nox.ClientApi.Tests/Tests/Controllers/WorkplacesControllerTests.cs:378]: Its this being used?

File: src/Nox.ClientApi.Tests/Tests/NoxWebApiTestBase.cs [hunk 4]
```diff
@@ -273,4 +273,11 @@ private void EnsureOdataSingleResponse(string content)
         var entity = ctx.Set<TResult>().Where(filter).FirstOrDefault();
         return entity;
     }
+    
+    protected List<TResult>? GetEntitiesByFilter<TResult>(Func<TResult, bool> filter) where TResult : class
```
Comment (rochar) [inline src/Nox.ClientApi.Tests/Tests/NoxWebApiTestBase.cs:277]: we need to discuss what is the goal of ClientApi.Tes, accessing directly to the DB, we need to think about this!

File: src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands/CreateWorkplaceTranslationsCommand.g.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+	{
+		cancellationToken.ThrowIfCancellationRequested();
+		await OnExecutingAsync(command);
+		command.WorkplaceLocalizedCreateDto.Id = command.Id;
+		command.WorkplaceLocalizedCreateDto.CultureCode = command.CultureCode;
+		var entityLocalizedToCreate = EntityLocalizedFactory.CreateLocalizedEntity(command.WorkplaceLocalizedCreateDto);
+		await OnCompletedAsync(command, entityLocalizedToCreate);
+		DbContext.WorkplacesLocalized.Add(entityLocalizedToCreate);
+		await DbContext.SaveChangesAsync();
+		return new WorkplaceLocalizedKeyDto(entityLocalizedToCreate.Id.Value, entityLocalizedToCreate.CultureCode.Value);
```
Comment (rochar) [inline src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Application.Commands/CreateWorkplaceTranslationsCommand.g.cs:65]: If its the default culture should also update the main entity

File: src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData/WorkplacesController/Translations.g.cs
Comment (rochar) [inline src/Nox.ClientApi.Tests/Generated/Nox.Generator/Nox.Generator.NoxCodeGenerator/Presentation.Api.OData/WorkplacesController/Translations.g.cs:1]: @ggrgovski  we miss to define the delete end point, or did we assume put null to be the same as deleted?

When this is completed we need to review the ticket for Get Translations make it consistent with any changes that we agree on this one

**Review by rochar — CHANGES_REQUESTED** [review]
looks good
some comment, please catch up with @ggrgovski

**Comment by msadikkoseaupa** [comment]
This pr won't be used any more, new brand new pr has been created regarding the latest design changes.

**Pull request closed without merging**